public/js/main.js

```diff
@@ -265,3 +257,16 @@ function copy(textareaId) {
     textarea.select();
     document.execCommand('copy');
 }
+
+function bytesToBase64(bytes) {
+    const binString = Array.from(bytes, (byte) =>
+        String.fromCodePoint(byte),
+    ).join("");
+    return btoa(binString);
+}
+
+async function CompressBlob(blob) {
+    const cs = new CompressionStream("gzip");
+    const compressedStream = blob.stream().pipeThrough(cs);
+    return await new Response(compressedStream).blob();
+}
```

Comment:
Compression using CompressionStream should fallback if its unsupported, since this api is still relatively new

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Compression_Streams_API#browser_compatibility
It looks like most browsers got it in 2020, and the stragglers got it in 2023... so I think the odds are pretty low...

Since it's also used on the decompression side... should the baked .html outputs optionally import [pako](https://github.com/nodeca/pako) if it can't find Compression/Decompression Streams?

Reply:
>I think the odds are pretty low

You'd be surprised :). I had several issues a while back when I broke support for a GPU that hasn't had updates for 5 years so it's definitely needed.

I would rather not pull in dependencies with the output code, though if you relied on zip compression then EmulatorJS does have the ability to detect and decompress that

Reply:
Hmm, I can switch it for `deflate`, and check to see if EmulatorJS can handle that directly... is there anything I need to do to let it know that it's compressed?   Perhaps a MIME type in the blob URL?

Reply:
Bah, CompressionStreams can natively produce `.gz` files, but not `.zip` files (which can be understood by EmulatorJS).

Luckily, it turns out there's a way to make JSZip actually run compression 😅  https://stackoverflow.com/a/41614764

So perhaps I'll discard this PR in exchange for using the existing libraries...